### PR TITLE
Document `this` variable in state-based template entities

### PR DESCRIPTION
**Pull request by emontnemery**
## Proposed change

Document `this` variable in state-based template sensors

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/65201
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/template.markdown [hunk 1]
```diff
@@ -67,6 +67,10 @@ template:
 
 {% endraw %}
 
+### Template variables
+
+State-based template sensors have the special template variable `this` available in their teemplates. The `this` variable aids [self-referencing](#self_referencing) of an entity's state and attribute in templates.
```
Comment (frenck) [inline source/_integrations/template.markdown]:
```suggestion
State-based template sensors have the special template variable `this` available in their templates. The `this` variable aids [self-referencing](#self_referencing) of an entity's state and attribute in templates.
```

File: source/_integrations/template.markdown [hunk 2]
```diff
@@ -561,6 +565,24 @@ template:
 
 {% endraw %}
 
+### Self referencing
+
+This example demonstrates how the `this` varibale can be used in templates for self referencing.
```
Comment (frenck) [inline source/_integrations/template.markdown]:
```suggestion
This example demonstrates how the `this` variable can be used in templates for self-referencing.
```

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Review by frenck — APPROVED** [review]

**Comment by Mariusthvdb** [comment]
on the docs, wouldn't  it be a good idea to add something less complex than basing the state on an attribute, but more the other way round, having the other config variables use the this variable from the state:

```
      - unique_id: season_name
        state: >
          {{states('sensor.season_astronomical').capitalize()}}
        name: >
          {{states('sensor.season_astronomical').capitalize()}}
        picture: >
          {% set state = states('sensor.season_astronomical') %}
          /local/season/{{state}}.png
```

would then become:
```
      - unique_id: season_name
        state: >
          {{states('sensor.season_astronomical')}}
        name: >
          {{this.state.capitalize()}}
        picture: >
          /local/season/{{this.state}}.png
```

or for icon:
```
      - unique_id: utc_offset
        name: Utc offset
        state: >
          {{now().utcoffset().total_seconds()/3600}}
        icon: >
          mdi:numeric-{{this.state|int|abs}}
```
instead of what we need to do now:
```
        icon: >
          {% set offset = now().utcoffset().total_seconds()/3600 %}
          mdi:numeric-{{offset|int|abs}}
```

**Pull request merged**